src/unstract/adapters/x2text/helper.py

```diff
@@ -3,14 +3,46 @@
 
 import requests
 from requests import Response
+from requests.exceptions import ConnectionError, HTTPError, Timeout
 
 from unstract.adapters.exceptions import AdapterError
+from unstract.adapters.utils import AdapterUtils
 from unstract.adapters.x2text.constants import X2TextConstants
 
 logger = logging.getLogger(__name__)
 
 
+class X2TextHelper:
+    """Helpers meant for x2text adapters."""
+
+    @staticmethod
+    def parse_response(
+        response: Response, out_file_path: Optional[str] = None
+    ) -> tuple[str, bool]:
+        """Parses the response from a request.
+
+        Optionally it can write the output to a file
+
+        Args:
+            response (Response): Response to parse
+            out_file_path (Optional[str], optional): Output file path to write
+                 to, skipped if None or emtpy. Defaults to None.
+        Returns:
+            tuple[str, bool]: Response's content and status of parsing
+        """
+        if not response.ok and not response.content:
```

Comment:
Here we are looking for explicit success. What about recording the actual error response that x2text-service would have sent? Don;t we have to bubble that up?

Reply:
Guess that is happening inside make_request()...